openpathsampling/pathmover.py

```diff
@@ -1602,18 +1778,73 @@ class OneWayShootingMover(RandomChoiceMover):
         valid ensembles; None implies all ensembles are allowed (no
         restriction)
     """
-    def __init__(self, selector, ensembles=None):
+    def __init__(self, ensemble, selector):
         movers = [
-            ForwardShootMover(selector, ensembles),
-            BackwardShootMover(selector, ensembles)
+            ForwardShootMover(
+                ensemble=ensemble,
+                selector=selector
+            ),
+            BackwardShootMover(
+                ensemble=ensemble,
+                selector=selector
+            )
         ]
         super(OneWayShootingMover, self).__init__(
-            movers=movers, ensembles=ensembles
+            movers=movers
         )
-        self.selector = selector
 
+    @classmethod
+    def from_dict(cls, dct):
+        mover = cls(
+            ensemble=dct['movers'][0].ensemble,
+            selector=dct['movers'][0].selector
+        )
+
+        # override with stored movers
+        mover.movers = dct['movers']
+
+        return mover
+
+class OneWayExtendMover(RandomChoiceMover):
```

Comment:
In principle yes. But I wrote it a little shorter. The current way first creates a new object with the correct ensemble and selector and then fixes it to use the stored movers. The new way does not instatiate using OneWayShooter, but creates an empty object using __new__ and then calls the super __init__ function to create a 'SelectionMover'. This is shorter and reuses more code.